src/java/org/apache/cassandra/index/sai/disk/v2/sortedterms/SortedTermsWriter.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk.v2.sortedterms;
+
+import java.io.IOException;
+
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import com.google.common.base.Preconditions;
+
+import org.apache.cassandra.index.sai.disk.io.IndexOutputWriter;
+import org.apache.cassandra.index.sai.disk.io.RAMIndexOutput;
+import org.apache.cassandra.io.tries.IncrementalDeepTrieWriterPageAware;
+import org.apache.cassandra.utils.bytecomparable.ByteComparable;
+import org.apache.cassandra.utils.bytecomparable.ByteSource;
+import org.apache.lucene.store.IndexOutput;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.BytesRefBuilder;
+import org.apache.lucene.util.StringHelper;
+import org.apache.lucene.util.packed.DirectMonotonicWriter;
+import org.apache.lucene.util.packed.PackedInts;
+import org.apache.lucene.util.packed.PackedLongValues;
+
+
+import static org.apache.cassandra.index.sai.disk.v1.trie.TrieTermsDictionaryReader.trieSerializer;
+
+/**
+ * Writes an ordered sequence of terms for use with {@link SortedTermsReader}.
+ * <p>
+ * Terms must be added in lexicographical ascending order.
+ * Terms can be of varying lengths.
+ *
+ * <p>
+ * Important implementation note: SAI blocked packed readers are slow,
+ * and Lucene MonotonicBlockPackedReader is slow.  Using them
+ * will cause this class to slow considerably.
+ *
+ * For documentation of the underlying on-disk data structures, see the package documentation.
+ *
+ * @see SortedTermsReader
+ * @see org.apache.cassandra.index.sai.disk.v2.sortedterms
+ */
+@NotThreadSafe
+public class SortedTermsWriter
+{
+    static final int TERMS_DICT_BLOCK_SHIFT = 4;
```

Comment:
It would be nice to have some description as to the meaning of these 4 constants.

Reply:
I agree, but that's from lucene, and honestly I don't really know what they are doing

Reply:
Ok, I think I've figured it out.